pages/settings/index.js

```diff
@@ -197,6 +220,42 @@ export default function Settings ({ ssrData }) {
                   }
                 }
               })
+
+              let signerInstanceKey = settings.signerInstanceKey
+              if (signer !== settings.signer || signerType !== settings.signerType) {
+                // if the signer changes, we regenerate the signerInstanceKey.
+                // this is used to identify the app instance by nip46 and permit
+                // token reuse
+                signerInstanceKey = bytesToHex(generateSecretKey())
+
+                // we create the initial connection for the nip46 signer here
+                // because it makes for better ux (no confirmation delay on crosspost)
+                // and because we can test immediately if the connection works
+                if (signerType === 'nip46' && signer) {
+                  const nostr = new Nostr()
+                  try {
+                    await callWithTimeout(async () => {
+                      const testSignerInstance = nostr.getSigner({
+                        userPreferences: { signer, signerType, signerInstanceKey }
+                      })
+                      if (testSignerInstance instanceof NDKNip46Signer) {
+                        testSignerInstance.once('authUrl', nostrAuthChallengeResolver)
+                      }
+                      await testSignerInstance.blockUntilReady()
```

Comment:
Here, we initiate and test the connection to the signer, providing two benefits:

1. The user immediately knows if the NIP-46 signer is functioning without needing to attempt a crosspost.
2. We complete the challenge and authentication at this stage, preventing interruptions during navigation later.